libs/input-mapping/src/Staging/ProviderInterface.php

```diff
@@ -1,23 +1,29 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Keboola\InputMapping\Staging;
 
 interface ProviderInterface
 {
-    /**
-     * @return string
-     */
-    public function getWorkspaceId();
+    public function getWorkspaceId(): string;
 
-    public function cleanup();
+    public function cleanup(): void;
 
     /**
-     * @return array
+     * @return array{
+     *      container?: string,
+     *      connectionString?: string,
+     *      host?: string,
+     *      warehouse?: string,
+     *      database?: string,
+     *      schema?: string,
+     *      user?: string,
+     *      password?: string,
+     *      account?: string,
```

Comment:
Nenabehli jsme si nekde nedavno, ze jsme meli pevne zavedenou strukturu credentials? Nedavalo by v tomhle pripade smysl to mit vedeny schvalne jen jako arbitrary array, protoze kazdy backend muz emit jinou strukturu credentials?

Reply:
no proto jsem dal všechno nullable a proto je to jen v phpstanu a ne objekt, takže nejhorší, co se může stát je, že ti zařve phpstan, že to nevrací field `newCredentialsField`, což mě přijde jako pokrok proti tomu, že vůbec nevíme co to pole může obsahovat

Reply:
Hmm.. ja se trochu bojim, ze kdyz ten novy field pribude a ja k tomu prijdu za 3 mesice, budu si zbytecne skrabat hlavu, jestli je blbe ten field, nebo ten docblock. Tazke bych prave volil spis cestu obecnyho `array` + klidne nejaky comment, ze je to backend specific a vic info se da dohledat nekde v dokumentaci (nebo spis neda :smile:). Ale to je takovy dost subjektivni duvod, blokovat to tim uricte nebudu.